### PR TITLE
PTX-13598 Add APIs for required objects to deploy PX Operator via marketplace

### PR DESCRIPTION
**Pull request by nikolaypopov**
Add APIs for required objects to deploy PX Operator via marketplace

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: k8s/operatormarketplace/operatormarketplace.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var (
+	instance Ops
+	once     sync.Once
+
+	deleteForegroundPolicy = metav1.DeletePropagationForeground
```
Comment (piyush-nimbalkar) [inline k8s/operatormarketplace/operatormarketplace.go:20]: Remove this if we are not using it.

File: k8s/operatormarketplace/operatormarketplace.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+// Ops is an interface to Operator operations.
+type Ops interface {
+	CatalogSourceOps
+	OperatorGroupOps
+	SubscriptionOps
+	InstallPlanOps
```
Comment (piyush-nimbalkar) [inline k8s/operatormarketplace/operatormarketplace.go:28]: ClusterServiceVersionOps is missing

Comment (nikolaypopov) [reply k8s/operatormarketplace/operatormarketplace.go:26]: Done

File: k8s/operatormarketplace/operatormarketplace.go (hunk 1 above)
Comment (nikolaypopov) [reply k8s/operatormarketplace/operatormarketplace.go]: Removed

File: k8s/operatormarketplace/catalogsource.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+func (c *Client) DeleteCatalogSource(name, namespace string) error {
+	if err := c.initClient(); err != nil {
+		return err
+	}
+
+	catalogSource := &v1alpha1.CatalogSource{}
+	catalogSource.Name = name
+	catalogSource.Namespace = namespace
+
+	if err := c.crClient.Delete(context.TODO(), catalogSource); err != nil {
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply k8s/operatormarketplace/catalogsource.go:98]: This works

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**